packages/next/src/server/lib/patch-fetch.ts

```diff
@@ -176,15 +172,15 @@ export function addImplicitTags(
 
   // Add the tags from the pathname. If the route has unknown params, we don't
   // want to add the pathname as a tag, as it will be invalid.
-  if (requestStore?.url.pathname && !hasFallbackRouteParams) {
-    const tag = `${NEXT_CACHE_IMPLICIT_TAG_ID}${requestStore.url.pathname}`
-    if (
-      !cacheStore ||
-      (cacheStore.type !== 'cache' && cacheStore.type !== 'unstable-cache')
-    ) {
-      if (!workStore.tags?.includes(tag)) {
-        workStore.tags?.push(tag)
-      }
+  if (
+    workUnitStore !== undefined &&
+    workUnitStore.type === 'request' &&
+    workUnitStore?.url.pathname &&
+    !hasFallbackRouteParams
+  ) {
+    const tag = `${NEXT_CACHE_IMPLICIT_TAG_ID}${workUnitStore.url.pathname}`
+    if (!workStore.tags?.includes(tag)) {
+      workStore.tags?.push(tag)
```

Comment:
Not sure if related to the revalidate tests failures but this doesn't seem semantically equivalent. previously we would still have a requestStore with a URL even during prerenders. Now we have that (though it should be removed) but we don't have access to that when adding implicit tags inside the patched fetch. It seems we may need to move this URL concept to workStore or add it to the prerender store.

Reply:
Now rebased on #70889